src/Microsoft.AspNetCore.SignalR.Specification.Tests/HubLifetimeManagerTestBase.cs

```diff
@@ -0,0 +1,142 @@
+                var connection1 = HubConnectionContextUtils.Create(client1.Connection);
+                var connection2 = HubConnectionContextUtils.Create(client2.Connection);
+
+                await manager.OnConnectedAsync(connection1).OrTimeout();
+                await manager.OnConnectedAsync(connection2).OrTimeout();
+
+                await manager.OnDisconnectedAsync(connection2).OrTimeout();
+
+                await manager.SendAllAsync("Hello", new object[] { "World" }).OrTimeout();
+
+                var message = Assert.IsType<InvocationMessage>(client1.TryRead());
+                Assert.Equal("Hello", message.Target);
+                Assert.Single(message.Arguments);
+                Assert.Equal("World", (string)message.Arguments[0]);
+
+                Assert.Null(client2.TryRead());
+            }
+        }
+
+        [Fact]
+        public async Task SendGroupAsyncWritesToAllConnectionsInGroupOutput()
+        {
+            using (var client1 = new TestClient())
+            using (var client2 = new TestClient())
+            {
+                var manager = CreateNewHubLifetimeManager();
+                var connection1 = HubConnectionContextUtils.Create(client1.Connection);
+                var connection2 = HubConnectionContextUtils.Create(client2.Connection);
+
+                await manager.OnConnectedAsync(connection1).OrTimeout();
+                await manager.OnConnectedAsync(connection2).OrTimeout();
+
+                await manager.AddToGroupAsync(connection1.ConnectionId, "gunit").OrTimeout();
+
+                await manager.SendGroupAsync("gunit", "Hello", new object[] { "World" }).OrTimeout();
+
+                var message = Assert.IsType<InvocationMessage>(client1.TryRead());
+                Assert.Equal("Hello", message.Target);
+                Assert.Single(message.Arguments);
+                Assert.Equal("World", (string)message.Arguments[0]);
+
+                Assert.Null(client2.TryRead());
+            }
+        }
+
+        [Fact]
+        public async Task SendGroupExceptAsyncDoesNotWriteToExcludedConnections()
+        {
+            using (var client1 = new TestClient())
+            using (var client2 = new TestClient())
+            {
+                var manager = CreateNewHubLifetimeManager();
+                var connection1 = HubConnectionContextUtils.Create(client1.Connection);
+                var connection2 = HubConnectionContextUtils.Create(client2.Connection);
+
+                await manager.OnConnectedAsync(connection1).OrTimeout();
+                await manager.OnConnectedAsync(connection2).OrTimeout();
+
+                await manager.AddToGroupAsync(connection1.ConnectionId, "gunit").OrTimeout();
+                await manager.AddToGroupAsync(connection2.ConnectionId, "gunit").OrTimeout();
+
+                await manager.SendGroupExceptAsync("gunit", "Hello", new object[] { "World" }, new[] { connection2.ConnectionId }).OrTimeout();
```

Comment:
Maybe now that we're going to "ship" these tests as things we want our customers to use we should just use boring old names like "Group1"... 😢

Reply:
LOL. @BrennanConroy made these names. 
I'll change them